### PR TITLE
Implement Balance method and corresponding tests

### LINKED ISSUE #8: Implement `Balance` method
[`Balance`](https://github.com/celestiaorg/go-cnc/blob/76fd1e3ad9062cdf9175b644a76778c8c0d92df3/client.go#L37) method needs to be implemented.

Specs: https://docs.celestia.org/developers/node-api/#get-balance

Type returned by celestia-node ([`Coin`](https://github.com/cosmos/cosmos-sdk/blob/61a97ef3475d04364fe7d1836dc7385f83f52b5b/proto/cosmos/base/v1beta1/coin.proto#L14) from cosmos-sdk) is so small, that it can be simply created locally, without introducing dependency on cosmos-sdk.

### PR DESCRIPTION
**Pull request by Regis-Caelum**
Signed-off-by: Regis-Caelum <khanmf@rknec.edu>

## Issues/PR

Fix Issue #8 

## Changes
- Added Balance `struct` according to [celestia documentation](https://docs.celestia.org/developers/node-api/#get-balance) on balance response.
- Implemented Balance method.
- Added Tests for Balance method.

Changes can be viewed in the files client.go, client_test.go, and type.go.

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -53,3 +53,9 @@ func TestSubmitPDF(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, txRes)
 }
+
+func TestBalance(t *testing.T) {
+	//t.Skip()
```
Comment (liamsi) [inline client_test.go:58]:
```suggestion
```

Comment (tzdybal) [reply client_test.go:58]: Just a note: #4 needs to be merged to enable tests.

File: client_test.go [hunk 2]
```diff
@@ -53,3 +53,9 @@ func TestSubmitPDF(t *testing.T) {
...
+	_, err := NewClient("http://localhost:26658", WithTimeout(30*time.Second))
```
[the remark this answers is not in the document]
Comment (tzdybal) [reply client_test.go:59]: Probably not. I used 30 seconds value because of `SubmitPFD` tests (it waits for celestia block).

File: types.go [hunk 3]
```diff
@@ -9,6 +9,11 @@ type SubmitPFDRequest struct {
 	GasLimit    uint64 `json:"gas_limit"`
 }
 
+type balanceRequest struct {
```
Comment (tzdybal) [inline types.go:12]:
```suggestion
type balanceResponse struct {
```

**Review by tzdybal — CHANGES_REQUESTED** [review]
Thanks a lot for your submission!

I left suggestions to rename `balanceRequest` as it's not a request.

I'll merge this PR after #4 is done, so tests can run automatically in CI.

File: client.go [hunk 4]
```diff
@@ -35,8 +35,23 @@ func (c *Client) Header(ctx context.Context, height uint64) /* Header */ error {
 }
 
 func (c *Client) Balance(ctx context.Context) error {
-	_ = balanceEndpoint
-	return errors.New("method Balance not implemented")
+	var result balanceRequest
```
[the remark this answers is not in the document]
Comment (Regis-Caelum) [reply client.go]: Fixed balance type

**Comment by tzdybal** [comment]
@Regis-Caelum please rebase or merge with `main`.

**Comment by Regis-Caelum** [comment]
merged the changes with main

**Pull request closed without merging**